Data_Structures_Answers.md

```diff
@@ -22,9 +22,23 @@ We drop the + 2 and the space complexity is O(1)
 
 
 5. What is the runtime complexity of the provided code in `names.py`?
+The runtime complexity of the provided code in names.py is O(n^2).
+For each item in Names_1, the inner loop runs 10,000 times. Thus 10,000 * 10,000.
 
 6. What is the space complexity of the provided code in `names.py`?
+The space complexity of the code provided in names.py is O(n^2).
+It creates one array for each list of names.
 
 7. What is the runtime complexity of your optimized code in `names.py`?
+The runtime complexity of the optimized code is as follows.
+First the algorithm sorts the names in list. This runtime complexity is O(log n)
+Next the algorithm goes through each name in names_2. The runtime for this is O(n). It has to check each name against the binary search tree.
+
+Therfore, the runtime for the total algorithm is O(n log n).
 
 8. What is the space complexity of your optimized code in `names.py`?
```

Comment:
aaaalmost Chad. Space complexity asks for how much _extra_ space you need when calling the function. so the space taken up by your input (in this case, the text files with the names) doesn't count here.